docs/query-classification-and-display.md

```diff
@@ -0,0 +1,101 @@
+# Query classification and display
+
+## Attributable Queries
+
+The results of some queries are unsuitable for attribution to individual
+developers. Most of them have a threshold value on which they trigger,
+for example all metric violations and statistics based queries. The
+results of such queries would all be attributed to the person pushing
+the value over (or under) the threshold. Some queries only trigger when
+another one doesn't. An example of this is the MaybeNull query which
+only triggers if the AlwaysNull query doesn't. A small change in the
+data flow could make an alert switch from AlwaysNull to MaybeNull (or
+vice versa). As a result we attribute both a fix and an introduction to
+the developer that changed the data flow. For this particular example
+the funny attribution results are more a nuisance than a real problem;
+the overall alert count remains unchanged. However, for the duplicate
+and similar code queries the effects can be much more severe, as they
+come in versions for "duplicate file" and "duplicate function" among
+many others, where "duplicate function" only triggers if "duplicate
+file" didn't. As a result adding some code to a duplicate file might
+result in a "fix" of a "duplicate file" alert and an introduction of
+many "duplicate function" alerts. This would be highly unfair.
+Currently, only the duplicate and similar code queries exhibit this
+"exchanging one for many" alerts when trying to attribute their results.
+Therefore we currently exclude all duplicate code related alerts from
+attribution.
+
+The following queries are excluded from attribution:
+
+- Metric violations, i.e. the ones with metadata properties like
+  `@(error|warning|recommendation)-(to|from)`
+- Queries with tag `non-attributable`
+
+This check is applied when the results of a single attribution are
+loaded into the datastore. This means that any change to this behaviour
+will only take effect on newly attributed revisions but the historical
+data remains unchanged.
+
+## Query severity and precision
+
+We currently classify queries on two axes, with some additional tags.
+Those axes are severity and precision, and are defined using the
+query-metadata properties `@problem.severity` and `@precision`.
+
+For severity, we have the following categories:
+
+- Error
+- Warning
+- Recommendation
+
+These categories may change in the future.
+
+For precision, we have the following categories:
+
+- Very-high
```

Comment:
Maybe use "Very high" (without hyphen) here since the tables below use this variant as well?

Reply:
Actually on closer inspection it should have the hyphen, and all of them shouldn't have a capital letter, as this is what is actually found in the query files. I've corrected this.